crates/network-scanner-net/src/async_io.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    num::NonZeroUsize,
+    os::windows::io::{AsRawSocket},
+    sync::Arc,
+};
+
+use polling::{Event, Events};
+use socket2::Socket;
+use tokio::sync::{Mutex};
+
+use crate::tokio_raw_socket::TokioRawSocket;
+
+#[derive(Debug)]
+pub struct AsyncIoRuntime {
+    poller: Arc<polling::Poller>,
+    socket_id: usize,
+    loop_handle: Option<std::thread::JoinHandle<()>>,
```

Comment:
question: What is the purpose of having this? If we want a way to shutdown the runtime, we need message passing to tell the polling thread to stop looping. It would be better to support that properly. A simple approach would be to have an `AtomicBool` that we check after the call to `wait`, and use [`notify`](https://docs.rs/polling/latest/polling/struct.Poller.html#method.notify) to make it return immediately.

note: Can be changed to `std::thread::JoinHandle<()>`, without the option if we spawn the thread eagerly at creation time.

Reply:
updated